CMake/HHVMExtensionConfig.cmake

```diff
@@ -605,6 +605,7 @@ function (HHVM_EXTENSION_INTERNAL_HANDLE_LIBRARY_DEPENDENCY extensionID dependen
     ${libraryName} STREQUAL "lz4" OR
     ${libraryName} STREQUAL "mbfl" OR
     ${libraryName} STREQUAL "mcrouter" OR
+    ${libraryName} STREQUAL "mysql" OR
```

Comment:
mysql is handled in hphp_link() in the exact same way as everything above and below, so I think it belongs here too?

The only part of the mysql branch below that was not already handled in HPHPFindLibs is setting -DENABLE_ASYNC_MYSQL=1 so I added it there, to have everything in one place.

Alternatively, should we remove all the mysql references from HPHPFindLibs and instead keep the mysql branch here (and only here)? That might be better if mysql is not referenced from anywhere else except these extensions...

Reply:
> The only part of the mysql branch below that was not already handled in HPHPFindLibs is setting -DENABLE_ASYNC_MYSQL=1 so I added it there, to have everything in one place.

We should probably just remove the option; the async extension is by far the best way to use MySQL with Hack, and I don't think we're testing (even building) with it disabled at all.

Either way, yeah, moving this to the specific extension seems like a good idea.

Reply:
I tested it, it builds and successfully runs a MySQL query :) But it seems to only support PHP's old `mysql_*` API which is deprecated so you're probably right, we should just kill it.

Curiously, ENABLE_ASYNC_MYSQL is only referenced from the *non*-async mysql extension.

The socket path (PHP_MYSQL_UNIX_SOCK_ADDR) is also only referenced from the non-async extension, as far as I can tell the async extension ignores it and uses the wrong path :( There aren't any references to any sockets in the async extension's code, I think it just delegates to squangle/libmysqlclient. Perhaps we need to pass the socket path to libmysqlclient instead?

> Either way, yeah, moving this to the specific extension seems like a good idea.

Alright, I'll try removing all the mysql stuff from HPHPFindLibs and leave it only in the extension code. Should I kill non-async mysql while I'm at it, or would we want to announce that first so people can prepare?

Reply:
> Curiously, ENABLE_ASYNC_MYSQL is only referenced from the non-async mysql extension.

> should I kill non-async mysql

This is a bit misleadingly named: its' primary purpose is/was to control if we build fbmysqlclient (or previously, webscalesqlclient)_

If we build fbmysqlclient:
- we can build the async extension
- the non-async extension also links against fbmysqlclient

If we don't:
- we can never build the async extension
- the non-async extension should link against a real libmysqlclient

I don't think we need to warn about removing support for linking against libmysqlclient if we keep the extension for now; are you also thinking of removing the non-async extension?

Reply:
Oh I think you meant removing the whole extension, if it's just the build option then yeah that should be fine. I'll test that + moving all the cmake stuff from HPHPFindLibs to the extension cmake config.